Detour/Source/DetourNavMeshQuery.cpp

```diff
@@ -1827,11 +1829,11 @@ dtStatus dtNavMeshQuery::findStraightPath(const float* startPos, const float* en
 					dtVcopy(portalApex, portalLeft);
 					apexIndex = leftIndex;
 					
-					unsigned char flags = 0;
+					dtStraightPathFlags flags;
 					if (!leftPolyRef)
-						flags = DT_STRAIGHTPATH_END;
+						flags.straightpathEnd = true;
```

Comment:
While `flags = DT_STRAIGHTPATH_END;` and `flags.straightpathEnd = true;` are indeed semantically different, in this context (where `flags` is initialized at line 1832) they're identical.